src/test/node.ts

```diff
@@ -51,6 +51,126 @@ describe("node", () => {
 			expect(messages.length).to.equal(1);
 			expect(messages).to.have.members(["hi"]);
 		});
+
+		it("it destroys the node stream and ends the ts-stream when the ts-stream's `write` call fails", async () => {
+			const errorMessage = "write failed";
+			const nodeStream = Readable.from(
+				(async function* () {
+					while (1) {
+						yield "hi";
+					}
+				})()
+			);
+
+			const stream = fromNodeReadable(nodeStream);
+
+			let ended: Error | undefined;
+			let aborted: Error | undefined;
+			await stream
+				.forEach(
+					// simulate a failing `write` call
+					(_chunk) => Promise.reject(new Error(errorMessage)),
+					(endErr) => {
+						ended = endErr;
+					},
+					(abortErr) => {
+						aborted = abortErr;
+					}
+				)
+				.catch((e) => e);
+
+			expect(nodeStream.destroyed).to.equal(true);
+			expect(ended?.message).to.equal(errorMessage);
+			expect(aborted).to.equal(undefined);
+		});
+
+		it("it destroys the node stream and ends the ts-stream when the ts-stream's `end` call fails", async () => {
+			const errorMessage = "end failed";
+			const nodeStream = Readable.from(
+				(async function* () {
+					yield "hi";
+				})()
+			);
+
+			const stream = fromNodeReadable(nodeStream);
+
+			let aborted: Error | undefined;
+			await stream
+				.forEach(
+					(_chunk) => Promise.resolve(),
+					// simulate a failing `end` call
+					(_endErr) => Promise.reject(new Error(errorMessage)),
+					(abortErr) => {
+						aborted = abortErr;
+					}
+				)
+				.catch((e) => e);
+
+			expect(nodeStream.destroyed).to.equal(true);
+			expect(aborted).to.equal(undefined);
```

Comment:
Add an additional assertion to verify that the "end failed" message was indeed the reason for ending things?